### PR TITLE
Thomas/first pass clinical features upload

### PR DESCRIPTION
**Pull request by ThomVett**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "react-measure": "^2.5.2",
     "react-router-dom": "^5.0.1",
     "react-scripts": "3.0.1",
+    "react-select": "^5.7.3",
```
Comment (roger-schaer) [inline package.json:48]:
```suggestion
```

Comment (roger-schaer) [reply package.json:48]: Maybe we don't need this dependency anymore?

File: src/Features.js [hunk 2]
```diff
@@ -724,6 +725,23 @@ function Features({ history }) {
...
                 </NavItem>
+                <NavItem>
+                  <NavLink
+                    className={getTabClassName('clinical_features')}
+                    onClick={() => {
+                      toggle('clinical_features');
+                    }}
+                  >
+                    {getTabSymbol()}
+                    {isAlternativeUser ? (
```
Comment (roger-schaer) [inline src/Features.js:736]: This test for an "alternative" user is not needed here, and might be obsolete overall (I will look into this in the future). There was initially an idea to have 2 version of the platform, one that includes visualization of features and one without, to make comparisons and see the benefit of visualizing the features. This idea wasn't realized in the end, and this test only applied to the Visualization tab in any case, as it needed to be called differently depending on the user group.

File: src/components/ClinicalFeatureTable.js [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    // "Gender": { "Type": CLINCAL_FEATURE_TYPES[0], "Encoding": "Categorical" }
+  });
+
+  const handleInputChange = (e, feature_name, feature_type) => {
+    let editableClinicalFeatureDefinitionsToUpdate = { ...editableClinicalFeatureDefinitions };
+    editableClinicalFeatureDefinitionsToUpdate[feature_name][feature_type] = e.target.value;
+    setEditableClinicalFeatureDefinitions(editableClinicalFeatureDefinitionsToUpdate)
+  };
+
+  useEffect(() => {
```
Comment (roger-schaer) [inline src/components/ClinicalFeatureTable.js:39]: By convention, it is usual to put all the "hooks" (useState, useEffect, etc.) at the top of the file, before declaring other functions, maybe you can hoist your useEffect hooks above the "handleInputChange" function?

File: src/components/ClinicalFeatureTable.js [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+
+    if (isValid) {
+      updateEditableClinicalFeatures(clinicalFeatures);
+      console.log(clinicalFeatures);
+    }
+    setisClinicalFeatureFileValid(isValid);
+    setclinicalFeatureFileMessage(message);
+  };
+
+  useEffect(() => {
```
Comment (roger-schaer) [inline src/components/ClinicalFeatureTable.js:115]: Same comment as for the other useEffect hook above

**Review by roger-schaer — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments and some changes we've already discussed (e.g. removing the "Manual Features" entry button), which can be updated at a later stage.

File: package.json (hunk 1 above)
Comment (ThomVett) [reply package.json:48]: no we do not - will remove

File: src/components/ClinicalFeatureTable.js (hunk 4 above)
Comment (ThomVett) [reply src/components/ClinicalFeatureTable.js:115]: works for me - don't know the react dev conventions yet!

**Pull request merged**